extensions/wikia/Wall/WallHooksHelper.class.php

```diff
@@ -529,86 +529,51 @@ static public function onSkinSubPageSubtitleAfterTitle( $title, &$ptext, &$cssCl
 	/**
 	 * @brief Adds an action button on user talk archive page
 	 *
-	 * @param $response
-	 * @param $ns
-	 * @param $skin
-	 * @return bool
+	 * @param array $button 'Edit' button and dropdown
+	 * @return bool true
 	 *
 	 * @author Andrzej 'nAndy' Łukaszewski
 	 */
-	static public function onPageHeaderIndexAfterActionButtonPrepared( $response, $ns, $skin ) {
+	static public function onPageHeaderIndexAfterActionButtonPrepared( array &$button ) {
 		$app = F::App();
 		$helper = new WallHelper();
 
 		if ( !empty( $app->wg->EnableWallExt ) ) {
 			$title = $app->wg->Title;
+			$ns = $title->getNamespace();
 			$parts = explode( '/', $title->getText() );
-			$action = $response->getVal( 'action' );
-			$dropdown = $response->getVal( 'dropdown' );
 			$canEdit = $app->wg->User->isAllowed( 'editwallarchivedpages' );
 
-			if ( $ns === NS_USER_WALL
+			if (
+				$ns === NS_USER_WALL
 				&& $title->isSubpage()
 				&& !empty( $parts[1] )
 				&& mb_strtolower( str_replace( ' ', '_', $parts[1] ) ) === mb_strtolower( $helper->getArchiveSubPageText() )
 			) {
-				// user talk archive
```

Comment:
I'm not 100% sure what I did here, but this conditional seems very close to the next one the only different being how the namespace is checked and that an extra argument is used when creating the edit URL. It also looks like the result of this conditional is immediately overridden by the next one - they set the same keys in the response.

At a guess, I think I spotted the duplication and removed this in favour of the later conditional as it appeared the later one was producing the result anyway.